app/models/product.rb

```diff
@@ -2,6 +2,15 @@ class Product < ApplicationRecord
   has_many :locations
   has_many :pallets
 
+  validates :client_id, presence: true
+  validates :code, presence: true, uniqueness: true
```

Comment:
all presence validations can be grouped in one line like this `validates_presence_of :first_name, :last_name, :birthday, :sex, :home`

Reply:
y can do it with an array?

Reply:
i'm not sure, try it before push the branch